### PR TITLE
Closes #1398: Adapt readonly mode on ui

### LINKED ISSUE #1398: Make readonly onstance work through api (native api)

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/builtin/DataverseUserPage.java [hunk 1]
```diff
@@ -239,10 +247,6 @@ public void changePassword(ActionEvent e) {
         editMode = EditMode.CHANGE_PASSWORD;
     }
 
-    public void forgotPassword(ActionEvent e) {
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/builtin/DataverseUserPage.java:242]: We can't forget password now :P?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/shib/ShibUtil.java [hunk 2]
```diff
@@ -184,8 +184,8 @@ static void mutateRequestForDevConstantHarvard1(HttpServletRequest request) {
          * Let's simulate the non-sending of displayName here.
          */
 //        request.setAttribute(displayNameAttribute, "John Harvard");
-        request.setAttribute(ShibUtil.firstNameAttribute, "John");
-        request.setAttribute(ShibUtil.lastNameAttribute, "Harvard");
+        request.setAttribute(ShibUtil.firstNameAttribute, "John2");
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/shib/ShibUtil.java]: What's the reason for a change?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java [hunk 3]
```diff
@@ -167,22 +175,23 @@ public boolean isRserveConfigured() {
     }
 
     public long getUploadLogoSizeLimit() {
-        return 500000;
+        return 500_000;
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: What kind of format is that :P? Why the change?

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/arquillian/arquillianexamples/WebappArquillianDeployment.java [hunk 4]
```diff
@@ -64,7 +66,8 @@ public void after() throws Throwable {
         JavaArchive javaArchive = ShrinkWrap.create(JavaArchive.class, "dv-webapp.jar")
                 .merge(PersistenceArquillianDeployment.createDeployment())
                 .addAsManifestResource(new FileAsset(new File("src/main/webapp/WEB-INF/beans.xml")), "beans.xml")
-                .addPackages(true, "edu.harvard.iq.dataverse");
+                .addPackages(true, "edu.harvard.iq.dataverse")
+                .deleteClass(WorkflowExecutionWorker.class);
```
Comment (MrK191) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/arquillian/arquillianexamples/WebappArquillianDeployment.java]: Hmm maybe some kind of javadoc why this class was deleted?

**Review by MrK191 — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/shib/ShibUtil.java (hunk 2 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/shib/ShibUtil.java]: It was just to check what will happen when attributes of user that comes from institutional login would change. I will revert it to original state

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/arquillian/arquillianexamples/WebappArquillianDeployment.java (hunk 4 above)
Comment (madryk) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/arquillian/arquillianexamples/WebappArquillianDeployment.java]: It is a leftover of my attempt to run integration tests in `dataverse-webapp` project. I will revert it as it should be done in separate task.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/builtin/DataverseUserPage.java (hunk 1 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/builtin/DataverseUserPage.java:242]: Yes - forgetting is illegal :)
Jokes aside:
Currently resetting password is done in separate page (`passwordreset.xhtml`)
So in `loginpage.xhtml` we have a direct link to `<a href="passwordreset.xhtml">`
When we enter valid email - then reset password url will be send to to our account (PasswordResetPage.sendPasswordResetLink()).
Url to reset password will be in form `http://<server>/passwordreset.xhtml?token=<token>`
And update of the password will be handled on passwordreset.xhtml page itself (PasswordResetPage.resetPassword())
This method seems to be some kind of leftover (maybe from how password resetting was handled in older versions of dataverse) - couldn't find any invocation of this method

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java (hunk 3 above)
Comment (madryk) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java]: https://docs.oracle.com/javase/7/docs/technotes/guides/language/underscores-literals.html
I've run pmd and it was one of the violations and I've changed it by mistake. I will get back to previous version

**Review by MrK191 — APPROVED** [review]
